### ISSUE TITLE
Illegal invocation error when retrying a payment with a different card

### ISSUE THREAD

**Issue by jamiembrown** [stripe/stripe-react-native#1171]
**Describe the bug**
Using `presentApplePay`, if the customer uses a card that fails, gets an error in the Apple Pay payment window (eg. card was declined), and then tries again, then `confirmApplePayPayment` crashes with the message:

```
Illegal callback invocation from native module. This callback type only permits a single invocation from native code.

RCTFatal
checkCallbackMultipleInvocations(bool*)
__41-[RCTModuleMethod processMethodSignature]_block_invoke_2.65
$syXlSgIeyBy_ypSgIegn_TR
$s19stripe_react_native9StripeSdkC15applePayContext_22didCreatePaymentMethod18paymentInformation10completiony0d5AppleG008STPApplegH0C_0D0010STPPaymentL0CSo9PKPaymentCySSSg_s5Error_pSgtctF
$s19stripe_react_native9StripeSdkC15applePayContext_22didCreatePaymentMethod18paymentInformation10completiony0d5AppleG008STPApplegH0C_0D0010STPPaymentL0CSo9PKPaymentCySSSg_s5Error_pSgtctFTo
$s6Stripe30STPApplePayContextLegacyHelperC29performDidCreatePaymentMethodyy0a5AppleC0013_stpinternal_lcdhijK7StorageCFZ
$s6Stripe30STPApplePayContextLegacyHelperC29performDidCreatePaymentMethodyy0a5AppleC0013_stpinternal_lcdhijK7StorageCFZTo
$s14StripeApplePay08STPAppleC7ContextC16_completePayment4with10completionySo9PKPaymentC_ySo0J19AuthorizationStatusV_s5Error_pSgtctFys6ResultOy0A4Core0A3APICAAE0G6MethodVsAK_pGcfU0_
$s10StripeCore12STPAPIClientC11sendRequest7request10completiony10Foundation10URLRequestV_ys6ResultOyxs5Error_pGctSeRzlFyAG4DataVSg_So13NSURLResponseCSgsAL_pSgtcfU_yyScMYccfU_
$sIeg_IeyB_TR
_dispatch_call_block_and_release
_dispatch_client_callout
_dispatch_main_queue_drain
_dispatch_main_queue_callback_4CF
42C5C917-0447-3995-B50F-DE4D132C2435
42C5C917-0447-3995-B50F-DE4D132C2435
CFRunLoopRunSpecific
GSEventRunModal
7B942FA4-CB76-3375-9972-F58C14492FB4
UIApplicationMain
main
341BBF64-6034-357E-8AA6-E1E4B988E03C
```

**To Reproduce**
1. Using this very basic code:

```
    const applePayResult = await presentApplePay({
      cartItems: [{ label: props.owner, amount: (props.amount / 100).toString(), paymentType: "Immediate" }],
      country: props.country,
      currency: props.currency,
      shippingMethods: [
        {
          amount: (props.shipping_price).toString(),
          identifier: 'Standard',
          label: 'Standard Shipping',
          detail: 'Estimated ' + props.shipping_days + ' Days',
          type: 'final',
        },
      ],
      requiredShippingAddressFields: ['emailAddress', 'phoneNumber', 'postalAddress', 'name'],
      requiredBillingContactFields: ['name', 'postalAddress'],
    });

    if (applePayResult.error) {
      console.log("Got error from presentApplePay")
      console.log(applePayResult.error.toString());
      props.source.setPaymentError(applePayResult.error.toString());
      }
      return;
    }

    props.source.setLoading();

    console.log("Got success from presentApplePay")

    applePayConfirmResult = await confirmApplePayPayment(
      props.intent.toString()
    );

    if (applePayConfirmResult.error) {
      console.log("Got error from confirmApplePayPayment")
      console.log(applePayConfirmResult.error);
      props.source.setPaymentError(applePayConfirmResult.error.toString());
      return;
    }
```

2. The Apple Pay form shows when the payment starts, and the customer submits with a card that doesn't work.
3. The customer gets an error in the Apple Pay window ("Payment failed! Try a different card"!) because the bank declined their card.
4. Customer tries another card, or retries the same card.
5. The app crashes in the background with the error above.

**Expected behavior**
Either Apple Pay should close (or we can close it) when the error occurs, or it should let them try as many times as they'd like.

**Screenshots**

1. First attempt is processing (white parts are personal information I've deleted):
![IMG_3523](https://user-images.githubusercontent.com/6493637/195703213-52a5abfa-8610-49f7-9a41-4de45ab33f85.PNG)

2. First attempt failed
![IMG_3524](https://user-images.githubusercontent.com/6493637/195703235-cc66545f-293c-4abe-8b23-b517651e650b.PNG)

3. Lets try again with the same card
![IMG_3525](https://user-images.githubusercontent.com/6493637/195703248-00d57a1a-5ec2-42a1-a082-9d2081799713.PNG)

4. Second attempt processing
![IMG_3526](https://user-images.githubusercontent.com/6493637/195703256-36b8bfdc-1bc9-407f-b318-d40eccd7d354.PNG)

5. Oops, app crashed in the background.
![IMG_3527](https://user-images.githubusercontent.com/6493637/195703265-ec553c8a-87af-4d5a-802d-655deda321b3.PNG)

**Smartphone (please complete the following information):**
 - Device: iPhone 11 Pro Max
 - OS: iOS 16.0.2
 - React Native: 8.0.6
 - Stripe: @stripe/stripe-react-native@0.19.0

**Additional context**
This is happening in live with a real card - so not a simiulator issue. Thank you for your help!

**Comment by shanereid** [comment]
+1, also seeing this issue with:
- Stripe 0.19.0
- iOS 16.0
- Reach Native: 0.70.3

**Comment by charliecruzan-stripe** [comment]
hi! this should be fixed with the new Platform Pay functions in v0.22.0, try using `confirmPlatformPayPayment` instead
